From 11aa6bce044e68df020b843ea9c350a048bd5e9d Mon Sep 17 00:00:00 2001
From: Sven-Ola Tuecke <sven-ola@gmx.de>
Date: Fri, 20 Feb 2026 14:55:43 -0500
Subject: [PATCH] Orange Pi RV2: u-boot support for XM25QU128C flash

NOR flash name from u-boot/master, ID from Xunlong Ky-Tree.

Signed-off-by: Sven-Ola Tuecke <sven-ola@gmx.de>
---
 configs/k1_defconfig          | 1 +
 drivers/mtd/spi/spi-nor-ids.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/configs/k1_defconfig b/configs/k1_defconfig
index 4bf6ee31..653c89d0 100644
--- a/configs/k1_defconfig
+++ b/configs/k1_defconfig
@@ -208,6 +208,7 @@ CONFIG_MTD_SPI_NAND=y
 CONFIG_SPI_NAND_GIGADEVICE=y
 CONFIG_SPI_FLASH_GIGADEVICE=y
 CONFIG_SPI_FLASH_WINBOND=y
+CONFIG_SPI_FLASH_XMC=y
 CONFIG_SPI_FLASH_FM=y
 CONFIG_SPINOR_BLOCK_SUPPORT=y
 # CONFIG_SPI_FLASH_USE_4K_SECTORS is not set
diff --git a/drivers/mtd/spi/spi-nor-ids.c b/drivers/mtd/spi/spi-nor-ids.c
index 855b1bf6..d8e7a527 100644
--- a/drivers/mtd/spi/spi-nor-ids.c
+++ b/drivers/mtd/spi/spi-nor-ids.c
@@ -455,6 +455,7 @@ const struct flash_info spi_nor_ids[] = {
 	{ INFO("XM25QH64A", 0x207017, 0, 64 * 1024, 128, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ INFO("XM25QH64C", 0x204017, 0, 64 * 1024, 128, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ INFO("XM25QH128A", 0x207018, 0, 64 * 1024, 256, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+	{ INFO("XM25QU128C", 0x204118, 0, 64 * 1024, 256, SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 #endif
 #ifdef CONFIG_SPI_FLASH_XTX
 	/* XTX Technology (Shenzhen) Limited */
-- 
2.51.0

